examples/front-proxy/service.py

```diff
@@ -12,5 +26,19 @@ def hello(service_number):
                                     socket.gethostname(),
                                     socket.gethostbyname(socket.gethostname())))
 
+@app.route('/trace/<service_number>')
+def trace(service_number):
+    headers = {}
+    # call service 2 from service 1
+    if int(os.environ['SERVICE_NAME']) == 1 :
+        for header in TRACE_HEADERS_TO_PROPAGATE:
+            if header in request.headers:
+                headers[header] = request.headers[header]
```

Comment:
That is true in the normal case. The idea behind Envoy doing the tracing for you is that the application does not have to do any tracing nor does it have to spend resources propagating the traces to a collector server. It eliminates the complexity of adding additional libraries for the purpose of tracing and dealing with all the additional code.

Secondly, the idea is that Envoy's presence is transparent to the application. The amount of time spent in Envoy should be minuscule (if its sizeable, then there is something wrong with Envoy design :) ).

This is the Envoy model btw. I am just adding examples to go with the model. :).